pkgs/misc/vscode-extensions/vscodeWithConfiguration.nix

```diff
@@ -0,0 +1,38 @@
+# wrapper over vscode to control extensions per project (extensions folder will be created in execution path)
+{ lib                             
+, writeShellScriptBin             
+, extensionsFromVscodeMarketplace 
+, vscodeDefault
+}:
+## User input
+{ vscode ? vscodeDefault                          
+# extensions to be symlinked into the project's extensions folder
+, nixExtensions        ? []                   
+# extensions to be copied into the project's extensions folder
+, mutableExtensions    ? []        
+, vscodeExtsFolderName ? ".vscode-exts"        
+}:
+let 
+  nixExtsDrvs = extensionsFromVscodeMarketplace nixExtensions;
+  mutExtsDrvs = extensionsFromVscodeMarketplace mutableExtensions;
+
+  #removed not defined extensions
+  rmExtensions =  lib.optionalString (nixExtensions++mutableExtensions != []) ''
+    find ${vscodeExtsFolderName} -mindepth 1 -maxdepth 1 ${lib.concatMapStringsSep " " (e : ''! -iname ${e.publisher}.${e.name}'') (nixExtensions++mutableExtensions)} -exec sudo rm -rf {} \;
+  '';
+  #copy mutable extension out of the nix store
+  cpExtensions = ''
+    ${lib.concatMapStringsSep "\n" (e : ''ln -sfn ${e}/share/vscode/extensions/* ${vscodeExtsFolderName}/'') nixExtsDrvs}
+    ${lib.concatMapStringsSep "\n" (e : ''
+      cp -a ${e}/share/vscode/extensions/${e.vscodeExtUniqueId} ${vscodeExtsFolderName}/${e.vscodeExtUniqueId}-${(lib.findSingle (ext: ''${ext.publisher}.${ext.name}'' == e.vscodeExtUniqueId) "" "m" mutableExtensions ).version}
```

Comment:
Which extensions have problems when their directory is immutable? Could we fix these extensions to avoid this copying?

Reply:
using this expression:
```
{ pkgs ? import <nixpkgs>{}}:
pkgs.vscode-with-extensions.override ({
    vscodeExtensions = pkgs.vscode-utils.extensionsFromVscodeMarketplace ([
    {
      name = "csharp";
      publisher = "ms-vscode";
      version = "1.21.5";
      sha256 = "031yx2kpjrw1b86y5nq110k9ywjgl26730ljddly05d8fxxham05";
    } 
    ]);
  })

will cuase this error in vscode:
```
Failed at stage: touchBeginFile
Error: EACCES: permission denied, mkdir '/nix/store/vnz8ja4qzmg4qpbr5am647djadclswsm-vscode-extensions-1.40.1/share/vscode/extensions/ms-vscode.csharp/.omnisharp'
```

and this error when trying to use nix-collect-garbage :
```
deleting '/nix/store/trash'
12428 store paths deleted, 664.94 MiB freed
error: cannot unlink '/nix/store/trash/4mnbcnp5fb8495pjlzwrfy24c9yk122n-vscode/extension-ms-vscode-csharp-1.21.5/share/vscode/extensions/ms-vscode.csharp/.omnisharp/1.34.5/license.md': Permission denied
```
im using darwin.